File: backend/app/schemas/portfolio.py
from typing import Optional
from datetime import datetime
from pydantic import BaseModel


class PortfolioItemCreate(BaseModel):
    title: str
    category: str
    description: Optional[str] = ""


class PortfolioItemUpdate(BaseModel):
    title: Optional[str] = None
    category: Optional[str] = None
    description: Optional[str] = None


class PortfolioItemRead(BaseModel):
    id: int
    title: str
    category: str
    image: Optional[str]
    description: Optional[str]
    createdAt: Optional[datetime]

    class Config:
        from_attributes = True

File: frontend/shared/types/portfolio.ts
export interface PortfolioItemCreate {
  title: string;
  category: string;
  description?: string | null;
}

export interface PortfolioItemUpdate {
  title?: string | null;
  category?: string | null;
  description?: string | null;
}

export interface PortfolioItemRead {
  id: number;
  title: string;
  category: string;
  image?: string | null;
  description?: string | null;
  createdAt?: string | null;
}